lib/Factory/LayoutFactory.php

```diff
@@ -288,6 +289,65 @@ public function getById($layoutId)
         return $layouts[0];
     }
 
+    /**
+     * Get CampaignId from layout history
+     * @param int $layoutId
+     * @return int campaignId|null
+     * @throws NotFoundException
+     */
+    public function getCampaignId($layoutId)
+    {
+        $campaignId = null;
+
+        if ($layoutId == 0)
+            throw new NotFoundException();
+
+        try {
+            $row = $this->getStore()->select('SELECT campaignId FROM `layouthistory` WHERE layoutId = :layoutId LIMIT 1', ['layoutId' => $layoutId]);
+            if(count($row) > 0) {
+                $campaignId = (int) $row[0]['campaignId'];
+            } else {
+                // layouthistory is not added on upgrade for a deleted layout but there is
+                // still stats exist in the player
+                $campaignId = null;
+                $this->getLog()->debug('Campaign ID not found for layoutId: '.$layoutId);
+            }
+        } catch (\RuntimeException $error) {
+            $this->getLog()->error($error);
+        }
+
+        // Set our Campaign ID
+        return $campaignId;
+    }
+
+    /**
+     * Get latest layoutId by CampaignId from layout history
+     * @param int campaignId
+     * @return int layoutId
+     * @throws NotFoundException
+     */
+    public function getLatestLayoutId($campaignId)
```

Comment:
Maybe we should call this `getLatestLayoutIdFromLayoutHistory()` just so it is clear where it goes for that information?